src/bundles/repl/functions.ts

```diff
@@ -1,68 +1,110 @@
- * @param {background_color_alpha} the alpha (transparency) of the original background color that covers on your background image [0 ~ 1]. Recommended value is 0.5 .
- *
- * @category Main
- */
-export function set_background_image(img_url: string, background_color_alpha: number) : void {
-  INSTANCE.customizedEditorProps.backgroundImageUrl = img_url;
-  INSTANCE.customizedEditorProps.backgroundColorAlpha = background_color_alpha;
-}
-
-
-/**
- * Set Programmable Repl editor font size
- * @param {font_size_px} font size (in pixel)
- *
- * @category Main
- */
-export function set_font_size(font_size_px: number) {
-  INSTANCE.customizedEditorProps.fontSize = parseInt(font_size_px.toString());// The TypeScript type checker will throw an error as "parseInt" in TypeScript only accepts one string as parameter.
-}
+/**
+ * Functions for Programmable REPL
+ * @module repl
+ * @author Wang Zihan
+ */
+
+
+import context from 'js-slang/context';
+import { ProgrammableRepl } from './programmable_repl';
+import { COLOR_REPL_DISPLAY_DEFAULT } from './config';
+
+const INSTANCE = new ProgrammableRepl();
+context.moduleContexts.repl.state = INSTANCE;
+/**
+ * Setup the programmable REPL with given metacircular evaulator entrance function
+ * @param {evalFunc} evalFunc - metacircular evaulator entrance function
+ *
+ * @category Main
+ */
+export function set_evaluator(evalFunc: Function) {
+  if (!(evalFunc instanceof Function)) {
+    const typeName = typeof (evalFunc);
+    throw new Error(`Wrong parameter type "${typeName}' in function "set_evaluator". It supposed to be a function and it's the entrance function of your metacircular evaulator.`);
+  }
+  INSTANCE.evalFunction = evalFunc;
+  return {
+    toReplString: () => '<Programmable Repl Initialized>',
+  };
+}
+
+
+/**
+ * Display message in Programmable Repl Tab
+ * If you give a pair as the parameter, it will use the given pair to generate rich text and use rich text display mode to display the string in Programmable Repl Tab with undefined return value (see module description for more information).
+ * If you give other things as the parameter, it will simply display the toString value of the parameter in Programmable Repl Tab and returns the displayed string itself.
+ *
+ * **Rich Text Display**
+ *  - First you need to `import { repl_display } from "repl";`
+ *  - Format: pair(pair("string",style),style)...
+ *  - Examples:
+ *
+ * `repl_display(pair(pair(pair(pair("Hello World", "underline"), "italic"), "bold"), "gigantic"));` (A large italic underlined "Hello World")
+ *
+ * `repl_display(pair(pair(pair(pair(pair("Hello World", "underline"),"italic"), "bold"), "gigantic"), "clrt#0000ff"));` (A large italic underlined "Hello World" in blue)
+ *
+ * `repl_display(pair(pair(pair(pair(pair(pair("Hello World", "underline"), "italic"), "bold"), "gigantic"), "clrb#A000A0"),"clrt#ff9700"));` (A large italic underlined "Hello World" with orange foreground and purple background)

```

Comment:
Just a minor comment, perhaps it would be better if we use the syntax-highlighted codeblocks instead of inline code? I.e.

````markdown
```js
// A large italic...
repl_display(...)
```
````

instead of

````markdown
`repl_display(...)`
````

Reply:
Done